src/Microsoft.Extensions.Logging.AzureAppServices/Internal/BatchingLoggerOptions.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.Logging.AzureAppServices.Internal
+{
+    public class BatchingLoggerOptions
```

Comment:
As mentioned before, can you clarify if `null` is valid? If it isn't make these types non-nullable, or don't throw when `!value.HasValue`